File: minimal_example/helloworld_client.cc
#include <grpcpp/grpcpp.h>
#include "simple_server.grpc.pb.h"

using grpc::Channel;
using grpc::ClientContext;
using grpc::Status;
using helloworld::HelloRequest;
using helloworld::HelloResponse;
using helloworld::HelloService;


class SimpleClient {

public: 
    SimpleClient(std::shared_ptr<Channel> channel) 
    : stub_(HelloService::NewStub(channel)) 
    {}

    std::string SayHello(const std::string& user) {
        HelloRequest request;
        request.set_greeting(user);

        HelloResponse reply;
        ClientContext context; 

        Status status = stub_->SayHello(&context, request, &reply);

        if (status.ok()) {
            return reply.reply();
        } else {
            std::cout << status.error_code() << ": " << status.error_message()
                      << std::endl;
            return "RPC failed";
        }
    }

private:
    std::unique_ptr<HelloService::Stub> stub_;
};

int main(int argc, char const *argv[])
{
    std::shared_ptr<Channel> channel = grpc::CreateChannel("localhost:9999", grpc::InsecureChannelCredentials());

    SimpleClient client(channel);

    std::string user("you c++");
    std::string reply = client.SayHello(user);

    std::cout << reply << std::endl;
    return 0;
}

File: minimal_example/helloworld_client.py
import grpc

import simple_server_pb2       # structures 
import simple_server_pb2_grpc  # stub 

if __name__ == "__main__":
    
    # create channel
    with grpc.insecure_channel('localhost:9999') as channel:
        # create stub with channel
        stub = simple_server_pb2_grpc.HelloServiceStub(channel)

        # make calls
        response = stub.SayHello(simple_server_pb2.HelloRequest(greeting='you'))
        print(response)

        responses = stub.SayManyHellos(simple_server_pb2.HelloRequest(greeting='me'))
        print(type(responses))
        for r in responses:
            print(r)